File: IDTECH/head_first_python/ch01_odd.py
from datetime import datetime
from os import getcwd
import time
import random

#odds = [1, 3, 5, 7, 9, 11, 13, 15, 17, 19, 21, 23, 25, 27, 29, 31, 33, 35, 37, 39, 41, 43, 45, 47, 51, 53, 55, 57,59 ]



#if right_this_minute in odds:
for i in range(1, 5):
    t = random.randint(1, 60)

    right_this_minute = datetime.today().minute
    
    print("> The current minute is: ", datetime.today().minute, "\n")
    
    if right_this_minute % 2 != 0: #used to replace line 3 and 6.
        print(">>> This minute seems a little odd.\n")
    else:
        print(">>> Not an odd minute. \n")
    print("...Now sleeping for:", t, "seconds\n")
    time.sleep(t)
which_dir = getcwd()
print("Your present working directory is: ", which_dir)

File: IDTECH/head_first_python/ch01_odd.c
#include <stdio.h>
#include <time.h>

int main() {

    time_t now = time(NULL); // get current time
    struct tm *t = localtime(&now); // convert to local time
    printf("Current time: %02d:%02d:%02d\n", t->tm_hour, t->tm_min, t->tm_sec);
    int right_this_minute = t->tm_min;
    if (right_this_minute % 2 != 0) {
        printf("This minute seems a little odd.\n");
        }
    else {
        printf("Not an odd minute.\n");
    }
    return 0;
}

// #include <stdio.h>
// #include <time.h>

// int main() {
//     time_t now = time(NULL); // get current time
//     struct tm *t = localtime(&now); // convert to local time
//     char time_str[9]; // buffer for formatted time string (HH:MM:SS\0)
//     strftime(time_str, sizeof(time_str), "%T", t); // format time as HH:MM:SS
//     printf("Current time of day: %s\n", time_str);
//     return 0;
// }
